src/reactors/RelayOrderReactor.sol

```diff
@@ -1,46 +1,42 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity ^0.8.0;
 
-import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
-import {ERC20} from "solmate/src/tokens/ERC20.sol";
 import {IPermit2} from "permit2/src/interfaces/IPermit2.sol";
 import {Permit2Lib} from "permit2/src/libraries/Permit2Lib.sol";
-import {ISignatureTransfer} from "permit2/src/interfaces/ISignatureTransfer.sol";
-import {SignedOrder} from "UniswapX/src/base/ReactorStructs.sol";
 import {ReactorEvents} from "UniswapX/src/base/ReactorEvents.sol";
-import {CurrencyLibrary} from "UniswapX/src/lib/CurrencyLibrary.sol";
 import {IRelayOrderReactor} from "../interfaces/IRelayOrderReactor.sol";
-import {ResolvedRelayOrder, RelayOrder} from "../base/ReactorStructs.sol";
+import {ResolvedTransferDetails, RelayOrder} from "../base/ReactorStructs.sol";
 import {ReactorErrors} from "../base/ReactorErrors.sol";
 import {Multicall} from "../base/Multicall.sol";
 import {RelayOrderLib} from "../lib/RelayOrderLib.sol";
-import {ResolvedRelayOrderLib} from "../lib/ResolvedRelayOrderLib.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {ActionsLib} from "../lib/ActionsLib.sol";
+import {SignedOrder} from "UniswapX/src/base/ReactorStructs.sol";
 
 /// @notice Reactor for handling the execution of RelayOrders
 /// @notice This contract MUST NOT have approvals or priviledged access
 /// @notice any funds in this contract can be swept away by anyone
 contract RelayOrderReactor is Multicall, ReactorEvents, ReactorErrors, IRelayOrderReactor {
-    using SafeTransferLib for ERC20;
-    using CurrencyLibrary for address;
-    using ResolvedRelayOrderLib for ResolvedRelayOrder;
     using RelayOrderLib for RelayOrder;
-    /// @notice permit2 address used for token transfers and signature verification
+    using ActionsLib for bytes[];
 
+    /// @notice permit2 address used for token transfers and signature verification
     IPermit2 public immutable permit2;
+    /// @notice Actions only execute on the universal router.
     address public immutable universalRouter;
 
     constructor(IPermit2 _permit2, address _universalRouter) {
         permit2 = _permit2;
         universalRouter = _universalRouter;
     }
 
-    /// @notice execute a signed RelayOrder
-    function execute(SignedOrder calldata order) external {
-        ResolvedRelayOrder memory resolvedOrder = resolve(order);
-
-        resolvedOrder.transferInputTokens(permit2);
-        executeActions(resolvedOrder.actions);
-        emit Fill(resolvedOrder.hash, msg.sender, resolvedOrder.swapper, resolvedOrder.permit.nonce);
+    // TODO: Consider adding nonReentrant.
+    function execute(SignedOrder calldata signedOrder) external returns (ResolvedTransferDetails memory resolved) {
+        (RelayOrder memory order) = abi.decode(signedOrder.order, (RelayOrder));
```

Comment:
or at least the same

Reply:
also think `Order` is maybe not the proper word here? theres no order, its more of just a transaction request / operation request

Reply:
`RelayRequest`? 

and yeah I was also surprised about the gas..

Reply:
So I dug into the gas implications... could it just be slight inaccuracies in how we benchmark gas. When we declare the interface to be `execute(RelayOrder calldata order, bytes calldata sig)` the test creates the RelayOrder struct in memory, the gas snapshot then starts but it includes copying the parameters from memory to calldata then inside the function it has to decode the calldata back to memory.

If instead we do `execute(RelayOrder memory order, bytes calldata sig)` gas decreases. But I think we want the param to be calldata... I just dont know how we can get accurate gas tests for that.

It is odd though because you'd think that something similar would be happening with the gas tests for the SignedOrder struct encoding/decoding but you can see [here](https://github.com/Uniswap/relayer/compare/flatten...eliminate-SignedOrder?expand=1) how I change it.. and gas goes up all around.